src/service_objects/game_piece_action_resolvers/attack_resolver.ts

```diff
@@ -0,0 +1,53 @@
+import * as Types from '../../graphql/__generated__/resolvers-types';
+import * as Models from '../../models/index.js';
+import { diceRoll } from '../../graphql/helpers';
+import { EncodedDiceRolls, ResolvedAttack } from '../../models/game_piece_action';
+
+export default function resolveAttacks(
+  attackingUnit: Models.Unit,
+  targetUnit: Models.Unit,
+  encodedDiceRolls: EncodedDiceRolls
+): ResolvedAttack[] {
```

Comment:
This function is pretty important to check out.

Reply:
Looks good to me, with the understanding that the dice rolls are more likely to be an array of array<3> rather than one long flat array.  I need to dig into that to confirm it though.